tests/unit/OptionsListUtilsTest.ts

```diff
@@ -948,34 +929,34 @@ describe('OptionsListUtils', () => {
-            const options = OptionsListUtils.getOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
-            const filteredOptions = OptionsListUtils.filterOptions(options, '998243aaaa');
+            const options = OptionsListUtils.getValidOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
+            const filteredOptions = OptionsListUtils.filterAndOrderOptions(options, '998243aaaa');
 
             expect(filteredOptions.recentReports.length).toBe(0);
             expect(filteredOptions.personalDetails.length).toBe(0);
             expect(filteredOptions.userToInvite).toBe(null);
         });
 
         it('should not return any options if search value does not match any personal details (getOptions)', () => {
-            const options = OptionsListUtils.getOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
-            const filteredOptions = OptionsListUtils.filterOptions(options, 'magneto');
+            const options = OptionsListUtils.getValidOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
+            const filteredOptions = OptionsListUtils.filterAndOrderOptions(options, 'magneto');
 
             expect(filteredOptions.personalDetails.length).toBe(0);
         });
 
         it('should return one recent report and no personal details if a search value provides an email (getOptions)', () => {
-            const options = OptionsListUtils.getOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
-            const filteredOptions = OptionsListUtils.filterOptions(options, 'peterparker@expensify.com', {sortByReportTypeInSearch: true});
+            const options = OptionsListUtils.getValidOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
+            const filteredOptions = OptionsListUtils.filterAndOrderOptions(options, 'peterparker@expensify.com', {sortByReportTypeInSearch: true});
             expect(filteredOptions.recentReports.length).toBe(1);
             expect(filteredOptions.recentReports.at(0)?.text).toBe('Spider-Man');
             expect(filteredOptions.personalDetails.length).toBe(0);
         });
 
         it('should return all matching reports and personal details (getOptions)', () => {
-            const options = OptionsListUtils.getOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails}, {maxRecentReportsToShow: 5});
-            const filteredOptions = OptionsListUtils.filterOptions(options, '.com');
+            const options = OptionsListUtils.getValidOptions({reports: OPTIONS.reports, personalDetails: OPTIONS.personalDetails});
+            const filteredOptions = OptionsListUtils.filterAndOrderOptions(options, '.com', {maxRecentReportsToShow: 5});
 
-            expect(filteredOptions.personalDetails.length).toBe(4);
+            expect(filteredOptions.personalDetails.length).toBe(2);
```

Comment:
I didn't get that why length changed from 4 to 2?

Reply:
Good catch, thanks! There was still something broken about the order and limit part. I fixed that now, reverted the changes in this test case and now the test case is passing again!